src/__tests__/resolveInlineRef.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { resolveInlineRef } from '../resolveInlineRef';
+
+describe('resolveInlineRef', () => {
+  test('should follow refs', () => {
+    const doc = {
+      a: {
+        $ref: '#/b/foo',
+      },
+      b: {
+        $ref: '#/c',
+      },
+      c: {
+        $ref: '#/d',
+      },
+      d: {
+        foo: 'woo!',
+      },
+    };
+
+    expect(resolveInlineRef(doc, '#/a')).toEqual('woo!');
+  });
+
+  test('should handle direct circular refs', () => {
+    const doc = {
+      a: {
+        $ref: '#/b',
+      },
+      b: {
+        $ref: '#/a',
+      },
+    };
+
+    expect(resolveInlineRef(doc, '#/a')).toEqual({
+      $ref: '#/b',
+    });
+  });
+
+  test('should handle direct circular refs #2', () => {
```

Comment:
This one is throwing me for a loop - am I thinking about it wrong?

First step, get `#/a`, yielding:

```json
{
  $ref: '#/b/foo',
}
```

Next get b/foo, yielding:

```json
{
  $ref: '#/c',
}
```

Next get c, yielding:

```json
{
  $ref: '#/b/bar',
}
```

Next get b/bar, yielding:

```json
{
  $ref: '#/e',
}
```

Next get b/bar, yielding:

```json
{
  $ref: '#/a',
}
```

Return this since as final value since we've already seen this value.

Reply:
Nope, you got it right. Good find. I need to tweak it.

Reply:
Approved, but one thought is should it just return itself it's circular? E.g. `$ref: '#/a'` rather than `$ref: '#/b/foo'`? Makes it easy for consumer to understand or check if it's circular, and kind of makes more sense?

Reply:
`itself` = the whole document or the last item we stopped upon?

Reply:
I made the change. LMK whether this is what you meant.